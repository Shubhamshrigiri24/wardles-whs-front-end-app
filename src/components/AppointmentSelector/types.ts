export interface Availability {
  resourceId: string;
  startTime: Date;
  endTime: Date;
}

export interface FetchAvailabilityInput {
  locationId?: string;
  serviceId: string;
  start: Date;
  end: Date;
}
