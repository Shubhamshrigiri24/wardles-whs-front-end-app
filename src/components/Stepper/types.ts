export enum StepperSteps {
  CONSULTATION = "Consultation",
  PRODUCT_SELECTION = "Treatment options",
  PERSONAL_DETAILS = "Personal details",
  DELIVERY_DETAILS = "Delivery details",
  PAYMENT_DETAILS = "Payment details",
}
