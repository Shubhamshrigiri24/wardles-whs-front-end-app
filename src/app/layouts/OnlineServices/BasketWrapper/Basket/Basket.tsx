import React, { FC, useCallback } from "react";
import { Typography, makeStyles, Theme, Grid } from "@material-ui/core";
import {
  Product,
  Pack,
  OnlineService,
} from "../../../../store/reducer/online/types";
import USPBanner from "components/USPBanner";

export const useStyles = makeStyles((theme: Theme) => ({
  container: {
    padding: theme.spacing(3),
    maxWidth: "372px",
    margin: "auto",
  },
  paddingLeft: {
    paddingLeft: theme.spacing(3),
  },
  paddingBottom: {
    paddingBottom: theme.spacing(4),
  },
  padding: {
    paddingTop: theme.spacing(2),
    paddingBottom: theme.spacing(4),
  },
  blueArea: {
    backgroundColor: "#E0ECFD",
    padding: theme.spacing(3),
    borderRadius: "8px",
    color: "#0061F2",
    display: "flex",
    flexDirection: "row",
    alignItems: "center",
  },
}));

export interface BasketProps {
  product?: Product;
  basketItems?: Pack[];
  onlineService: OnlineService;
}

const Basket: FC<BasketProps> = ({ product, basketItems, onlineService }) => {
  const classes = useStyles();

  const calculateTotal = useCallback(() => {
    return basketItems?.reduce((acc, curr) => {
      if (curr?.price) acc += curr.price;
      return acc;
    }, 0);
  }, [basketItems]);

  return (
    <div className={classes.container}>
      <Grid container className={classes.paddingBottom}>
        <Typography>
          <b>Your basket</b>
        </Typography>
        {basketItems?.map((item) => {
          return (
            <Grid
              key={`basketItem-${item?.label}`}
              item
              xs={12}
              className={classes.padding}
            >
              <Typography variant={"subtitle1"}>
                {`${product?.name ?? ""} ${product?.variant ?? ""}`}
              </Typography>

              <Typography variant={"subtitle1"}>{item?.label || ""}</Typography>
            </Grid>
          );
        })}
        <Grid item xs={6}>
          <Typography variant={"subtitle1"}>Total</Typography>
        </Grid>
        <Grid item xs={6}>
          <Typography variant={"subtitle1"} align={"right"}>
            Â£{calculateTotal()?.toFixed(2)}
          </Typography>
        </Grid>
      </Grid>
      {product !== undefined ? <USPBanner product={product} /> : null}
    </div>
  );
};

export default Basket;
