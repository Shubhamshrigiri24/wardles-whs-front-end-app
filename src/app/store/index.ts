import { combineReducers, createStore, applyMiddleware, compose } from "redux";
import { apiMiddleware } from "redux-api-middleware";
import thunk from "redux-thunk";

import { persistStore, persistReducer } from "redux-persist";
import storage from "redux-persist/lib/storage";

import checkoutReducer from "./reducer/checkout";
import onlineReducer from "./reducer/online";
import orderService from "./reducer/order";
import accountService from "./reducer/account";
import subscriptions from "./reducer/subscriptions";
import orders from "./reducer/orders";

const persistConfig = {
  key: "root",
  storage,
  blacklist: ["checkout", "online", "subscriptions", "orders"],
};

const checkoutPersistConfig = {
  key: "checkout",
  storage,
  blacklist: ["braintreeToken", "confirmedPayment"],
};

const orderPersistConfig = {
  key: "order",
  storage,
};

const accountPersistConfig = {
  key: "account",
  storage,
};

const checkout = persistReducer(checkoutPersistConfig, checkoutReducer);
const order = persistReducer(orderPersistConfig, orderService);
const account = persistReducer(accountPersistConfig, accountService);

const online = persistReducer(
  {
    key: "online",
    blacklist: ["onlineServices", "productTokenValid"],
    storage,
  },
  onlineReducer
);

const rootReducers = combineReducers({
  checkout,
  online,
  order,
  account,
  subscriptions,
  orders,
});

const persistedReducer = persistReducer(persistConfig, rootReducers);

declare global {
  interface Window {
    __REDUX_DEVTOOLS_EXTENSION_COMPOSE__?: typeof compose;
  }
}

const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;

export const store = createStore(
  persistedReducer,
  composeEnhancers(applyMiddleware(apiMiddleware, thunk))
);

export const persistor = persistStore(store);
