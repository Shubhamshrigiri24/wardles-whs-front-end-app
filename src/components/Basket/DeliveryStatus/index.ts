export { default, DeliveryStatus } from "./DeliveryStatus";
