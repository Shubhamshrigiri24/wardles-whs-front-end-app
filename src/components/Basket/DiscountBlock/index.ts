export { default, DiscountBlock } from "./DiscountBlock";
