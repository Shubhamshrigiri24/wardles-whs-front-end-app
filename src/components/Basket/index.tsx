import React, { FC, useState, useCallback } from "react";
import { useLocation, generatePath } from "react-router-dom";
import { useSelector } from "react-redux";
import { Typography, Spacing } from "@welldigital/components";
import clsx from "clsx";
import {
  Card,
  CardContent,
  Grid,
  Collapse,
  useMediaQuery,
  Box,
  Divider,
} from "@material-ui/core";
import { useTheme } from "@material-ui/core/styles";
import ShoppingCartIcon from "@material-ui/icons/ShoppingCart";
import LocalShippingIcon from "@material-ui/icons/LocalShipping";

import ExpandMore from "@material-ui/icons/ExpandMore";
import { formatPrice } from "utils/formatters";
import { OrderProduct } from "app/store/reducer/order/types";
import { getProduct } from "app/store/reducer/order/selectors";
import { RootState } from "app/store/types";
import { DiscountBlock } from "components/Basket/DiscountBlock";
import { DeliveryStatus } from "components/Basket/DeliveryStatus";
import { DELIVERY_LABELS_BY_TYPE } from "constants/product";
import { ORDER_PAYMENT_PATH, ACCOUNT_DETAILS_PATH } from "constants/paths";
import { useStyles } from "components/Basket/styles";

type BasketProps = {
  shouldSuppressDeliveryInfo?: boolean;
};

const Basket: FC<BasketProps> = ({ shouldSuppressDeliveryInfo = false }) => {
  const classes = useStyles();
  const location = useLocation();
  const theme = useTheme();
  const isDesktop = useMediaQuery(theme.breakpoints.up("sm"), { noSsr: true });
  const [isOpen, setIsOpen] = useState<boolean>(false);
  const product: OrderProduct = useSelector((state: RootState) => {
    return getProduct(state || {});
  });
  const hasDiscounts =
    product?.packs[0]?.discounts && product?.packs[0]?.discounts.length > 0;

  const deliveryType: keyof typeof DELIVERY_LABELS_BY_TYPE | undefined =
    product?.packs[0]?.delivery?.type;

  const isDetailsPage =
    location.pathname ===
    generatePath(ACCOUNT_DETAILS_PATH, { onlineServiceId: "ed" });

  const isPaymentPage =
    location.pathname ===
    generatePath(ORDER_PAYMENT_PATH, { onlineServiceId: "ed" });

  const shouldAddDelivery =
    (isDetailsPage || isPaymentPage) && !shouldSuppressDeliveryInfo;

  const handleCollapse = useCallback(() => {
    setIsOpen(!isOpen);
  }, [isOpen]);

  const renderEmptyState = () => (
    <Typography
      variant={"subtitle2"}
      spacingAfter={0}
      classes={{ root: classes.detailsText }}
    >
      You must select a product.
    </Typography>
  );
  const renderProduct = (productPaylod: OrderProduct) => (
    <>
      <Box paddingBottom={2} className={classes.productSummary}>
        <Grid container wrap={"nowrap"}>
          <img
            className={classes.image}
            src={productPaylod?.image as string}
            alt={"product"}
          />

          <Grid item style={{ flex: 1 }}>
            <Grid container justify={"space-between"} wrap={"nowrap"}>
              <Typography className={classes.defaultText}>
                {productPaylod?.name} {productPaylod?.variant},{" "}
                {productPaylod?.packs[0].label}
              </Typography>
              <Typography
                className={classes.defaultText}
                style={{ paddingLeft: "10px" }}
              >
                {formatPrice(productPaylod?.packs[0].basePrice ?? 0)}
              </Typography>
            </Grid>
            {productPaylod?.subscription &&
            productPaylod?.packs?.[0]?.discounts &&
            productPaylod?.packs?.[0]?.discounts.length === 0 ? (
              <Typography className={classes.defaultText}>
                Monthly subscription
              </Typography>
            ) : (
              <Spacing spacing={1} />
            )}
          </Grid>
        </Grid>
      </Box>
      <Box>
        <DiscountBlock />
      </Box>

      {/*@ts-ignore */}
      {shouldAddDelivery && productPaylod?.packs[0]?.delivery && (
        <>
          {!hasDiscounts && isPaymentPage ? null : (
            <>
              <Spacing spacing={3} />
              <Divider />
              <Spacing spacing={3} />
            </>
          )}
          <DeliveryStatus
            label={DELIVERY_LABELS_BY_TYPE[deliveryType!]}
            value={
              deliveryType !== "FREE"
                ? //@ts-ignore
                  formatPrice(productPaylod?.packs[0]?.delivery.price)
                : "Free"
            }
            Icon={LocalShippingIcon}
            uniform
          />
        </>
      )}

      <Box marginTop={3}>
        <Divider classes={{ root: classes.dividerDark }} />
      </Box>

      <Box className={classes.totalWrapper}>
        <Grid container spacing={1} alignItems={"center"}>
          <Grid item xs={6}>
            <Typography
              variant={"h4"}
              spacingAfter={0}
              classes={{ root: classes.darkColor }}
              style={{ fontWeight: 700 }}
            >
              Total to pay
            </Typography>
          </Grid>
          <Grid item xs={6}>
            <Typography
              variant={"h2"}
              align={"right"}
              spacingAfter={0}
              classes={{
                root: clsx(classes.darkColor, classes.totalPrice),
              }}
            >
              {formatPrice(
                (productPaylod?.packs[0].price ?? 0) +
                  (shouldAddDelivery
                    ? // @ts-ignore
                      productPaylod?.packs[0]?.delivery?.price ?? 0
                    : 0)
              )}
            </Typography>
          </Grid>
        </Grid>
      </Box>
    </>
  );

  return (
    <Box>
      <Card classes={{ root: classes.card }}>
        <CardContent classes={{ root: classes.cardContent }}>
          <Collapse
            in={isDesktop ? true : isOpen}
            timeout={"auto"}
            collapsedHeight={"32px"}
          >
            <Box
              className={clsx(classes.titleWrapper, {
                [classes.hasAction]: !isDesktop,
              })}
              onClick={!isDesktop ? handleCollapse : undefined}
            >
              <Box display={"flex"} alignItems={"center"}>
                <ShoppingCartIcon className={classes.titleIcon} />
                <Typography
                  variant={"h3"}
                  classes={{ root: classes.darkColor }}
                >
                  Basket summary
                </Typography>
              </Box>
              {!isDesktop && (
                <ExpandMore
                  className={clsx(classes.expand, {
                    [classes.expandOpen]: isOpen,
                  })}
                />
              )}
            </Box>
            {product && renderProduct(product)}
            {!product && renderEmptyState()}
          </Collapse>
        </CardContent>
      </Card>
    </Box>
  );
};

export default Basket;
