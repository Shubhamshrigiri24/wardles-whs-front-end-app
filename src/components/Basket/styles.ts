import { makeStyles, Theme } from "@material-ui/core";
import { wellColors } from "@welldigital/components";

export const useStyles = makeStyles((theme: Theme) => ({
  card: {
    border: `2px solid ${wellColors.greyscale[500]}`,
    borderRadius: "10px",
    [theme.breakpoints.up("sm")]: {
      border: "none",
      borderRadius: 0,
      borderLeft: `2px solid ${wellColors.greyscale[500]}`,
    },
  },
  cardContent: {
    padding: "28px 24px",
    display: "flex",
    flexDirection: "column",
    [theme.breakpoints.up("sm")]: {
      padding: theme.spacing(2, 0, 2, 3),
      "&:last-child": {
        paddingBottom: 0,
      },
    },
    [theme.breakpoints.up("md")]: {
      padding: theme.spacing(2, 6),
    },
  },
  titleWrapper: {
    marginBottom: "20px",
    display: "flex",
    alignItems: "center",
    justifyContent: "space-between",
  },
  hasAction: {
    cursor: "pointer",
  },
  titleIcon: {
    marginRight: "10px",
    color: wellColors.elixir[900],
  },
  expand: {
    transform: "rotate(0deg)",
    transition: theme.transitions.create("transform", {
      duration: theme.transitions.duration.shortest,
    }),
  },
  expandOpen: {
    transform: "rotate(180deg)",
  },
  promoIcon: {
    marginRight: "12px",
    color: wellColors.elixir[300],
  },
  image: {
    height: "60px",
    display: "block",
    marginRight: theme.spacing(2),
  },
  defaultText: {
    color: wellColors.elixir[600],
    fontWeight: 600,
  },
  detailsText: {
    color: wellColors.elixir[300],
    fontSize: "16px",
    fontWeight: 400,
  },
  // blueText: {
  //   color: wellColors.zen[400],
  //   fontWeight: 600,
  // },
  divider: {
    backgroundColor: wellColors.greyscale[300],
  },
  dividerDark: {
    backgroundColor: wellColors.elixir[400],
  },
  darkColor: {
    color: wellColors.elixir[900],
  },
  caption: {
    fontSize: "12px",
    fontWeight: "normal",
    lineHeight: "16px",
    color: wellColors.elixir[400],
    letterSpacing: "0.5px",
  },

  totalWrapper: {
    paddingTop: "12px",
    borderTop: `2px solid ${wellColors.elixir[900]}`,
  },

  totalPrice: {
    fontSize: "34px",
  },

  discountRow: {
    display: "flex",
    justifyContent: "space-between",
  },

  discountValue: {
    display: "flex",
    alignItems: "center",
    color: wellColors.elixir[200],
  },

  discountHeading: {
    color: wellColors.elixir[900],
    fontSize: "18px",
    fontWeight: 400,
  },

  discountMessage: {
    display: "flex",
    fontSize: "12px",
    fontWeight: 400,
  },

  subtotalWrapper: {
    paddingTop: theme.spacing(1),
    margin: theme.spacing(2, 0),
    borderTop: "1px solid #E6E6E6",
  },

  discountSubtotal: {
    fontSize: "18px",
    fontWeight: 700,
    color: wellColors.elixir[600],
  },

  errorMessage: {
    color: wellColors.bloom[800],
    fontSize: "14px",
    fontWeight: 600,
  },

  successMessage: {
    color: wellColors.tonic[800],
    fontSize: "14px",
    fontWeight: 600,
  },

  messageWrapper: {
    padding: theme.spacing(0, 2),
  },

  discountRemoveIcon: {
    color: wellColors.bloom[900],
    cursor: "pointer",
  },

  iconPlaceholder: {
    width: "20px",
  },

  productSummary: {
    borderBottom: "1px solid #E6E6E6",
  },
}));
