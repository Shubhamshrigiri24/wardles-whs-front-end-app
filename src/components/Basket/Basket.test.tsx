import React from "react";
import { render } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import { Router, generatePath } from "react-router-dom";
import { createMemoryHistory } from "history";
import userEvent from "@testing-library/user-event";
import { OrderProduct } from "app/store/reducer/order/types";
import { ORDER_PAYMENT_PATH } from "constants/paths";
import * as reactRedux from "react-redux";
import { store } from "app/store";
import Basket from "./index";

const product: OrderProduct = {
  id: "sildenafil",
  name: "Sildenafil",
  service: "ed",
  packs: [
    {
      isBestValue: false,
      itemsPerPack: 16,
      label: "16 tablets",
      name: "Sildenafil",
      price: 49.99,
      pricePerUnit: 3.12,
      sku: "SI05359848",
      basePrice: 123,
      discounts: [
        {
          code: "subscription",
          type: "percentage",
          discountInfo: 10,
          discountAmount: 5.2,
          priceWithDiscount: 46.79,
        },
      ],
    },
  ],
  variant: "50 mg",
  subscription: true,
  image: "/static/media/sildenafil.3523fa82.png",
};

const productWithDiscount: OrderProduct = {
  ...product,
  packs: [
    {
      ...product.packs[0],
      price: 44.99,
      discounts: [
        {
          code: "DUMMY_CODE_10",
          type: "percentage",
          discountAmount: 5,
          discountInfo: 10,
          priceWithDiscount: 40,
        },
      ],
    },
  ],
};

const history = createMemoryHistory();
const useSelectorMock = jest.spyOn(reactRedux, "useSelector");

beforeEach(() => {
  useSelectorMock.mockClear();
  history.push("/");
});

const renderWithStore = () =>
  render(
    <Router history={history}>
      <reactRedux.Provider store={store}>
        <Basket />
      </reactRedux.Provider>
    </Router>
  );

describe("<Basket />", () => {
  describe("When the component is rendered without a product", () => {
    it('should display a "missing product" message', () => {
      const { getByText } = renderWithStore();
      expect(getByText(/You must select a product/)).toBeInTheDocument();
    });

    it("should not display the discounts block", () => {
      const { queryByRole } = renderWithStore();
      expect(queryByRole("button", { name: "Apply discount" })).toBeNull();
    });
  });

  describe("When the component is rendered with a product", () => {
    it("Should display the discounts block", () => {
      useSelectorMock.mockReturnValue(product);
      const { getByRole } = renderWithStore();

      expect(
        getByRole("button", { name: "Apply discount" })
      ).toBeInTheDocument();
    });

    describe("When a valid discount code has been submitted", () => {
      it("should be displayed in the basket", () => {
        useSelectorMock.mockReturnValue(productWithDiscount);
        const { getByText } = renderWithStore();

        expect(getByText(/DUMMY_CODE_10: 10% off/)).toBeInTheDocument();
        expect(getByText(/£5.00/)).toBeInTheDocument();
      });

      it("should update the basket total", () => {
        useSelectorMock.mockReturnValue(productWithDiscount);
        const { getByText } = renderWithStore();

        expect(getByText(/£44.99/)).toBeInTheDocument();
      });

      describe("When the user tries to submit a second discount code", () => {
        it("should display error message", () => {
          useSelectorMock.mockReturnValue(productWithDiscount);
          const { getByTestId, getByRole, getByText } = renderWithStore();

          const inputField = getByTestId("basket/discount-code-input");
          const submitButton = getByRole("button", { name: "Apply discount" });

          userEvent.type(inputField, "SECOND_DISCOUNT_CODE_10");
          userEvent.click(submitButton);

          expect(
            getByText(/Only one discount code can be applied to your basket./)
          ).toBeInTheDocument();
        });
      });

      describe("When the user tries to submit an empty discount code", () => {
        it("should display error message", () => {
          useSelectorMock.mockReturnValue(product);
          const { getByRole, getByText } = renderWithStore();

          const submitButton = getByRole("button", { name: /Apply discount/ });

          userEvent.click(submitButton);

          expect(
            getByText(/Invalid discount code. Please try again/)
          ).toBeInTheDocument();
        });
      });

      describe("When the user is on payment page", () => {
        it("should disable discount codes interaction", () => {
          const paymentPageRoute = generatePath(ORDER_PAYMENT_PATH, {
            onlineServiceId: "ed",
          });
          history.push(paymentPageRoute);

          const { queryByRole, queryByTestId } = renderWithStore();

          expect(queryByRole("button", { name: /Apply discount/ })).toBeNull();
          expect(queryByTestId("basket/remove-1")).toBeNull();
        });
      });
    });

    it("should match snapshot", () => {
      useSelectorMock.mockReturnValue(product);
      const { container } = renderWithStore();
      expect(container).toMatchSnapshot();
    });
  });
});
