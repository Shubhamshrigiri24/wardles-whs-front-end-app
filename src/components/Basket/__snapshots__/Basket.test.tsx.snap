// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Basket /> When the component is rendered with a product should match snapshot 1`] = `
<div>
  <div
    class="MuiBox-root MuiBox-root-633"
  >
    <div
      class="MuiPaper-root MuiCard-root makeStyles-card-604 MuiPaper-elevation1 MuiPaper-rounded"
    >
      <div
        class="MuiCardContent-root makeStyles-cardContent-605"
      >
        <div
          class="MuiCollapse-container"
          style="min-height: 32px;"
        >
          <div
            class="MuiCollapse-wrapper"
          >
            <div
              class="MuiCollapse-wrapperInner"
            >
              <div
                class="MuiBox-root MuiBox-root-634 makeStyles-titleWrapper-606 makeStyles-hasAction-607"
              >
                <div
                  class="MuiBox-root MuiBox-root-635"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root makeStyles-titleIcon-608"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M7 18c-1.1 0-1.99.9-1.99 2S5.9 22 7 22s2-.9 2-2-.9-2-2-2zM1 2v2h2l3.6 7.59-1.35 2.45c-.16.28-.25.61-.25.96 0 1.1.9 2 2 2h12v-2H7.42c-.14 0-.25-.11-.25-.25l.03-.12.9-1.63h7.45c.75 0 1.41-.41 1.75-1.03l3.58-6.49c.08-.14.12-.31.12-.48 0-.55-.45-1-1-1H5.21l-.94-2H1zm16 16c-1.1 0-1.99.9-1.99 2s.89 2 1.99 2 2-.9 2-2-.9-2-2-2z"
                    />
                  </svg>
                  <h3
                    class="MuiTypography-root makeStyles-darkColor-617 MuiTypography-h3"
                  >
                    Basket summary
                  </h3>
                  <div
                    class="makeStyles-spacing4-644 makeStyles-spacing4Responsive-645"
                    data-testid="spacing/root"
                  />
                </div>
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root makeStyles-expand-609"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
                  />
                </svg>
              </div>
              <div
                class="MuiBox-root MuiBox-root-662 makeStyles-productSummary-632"
              >
                <div
                  class="MuiGrid-root MuiGrid-container MuiGrid-wrap-xs-nowrap"
                >
                  <img
                    alt="product"
                    class="makeStyles-image-612"
                    src="/static/media/sildenafil.3523fa82.png"
                  />
                  <div
                    class="MuiGrid-root MuiGrid-item"
                    style="flex: 1;"
                  >
                    <div
                      class="MuiGrid-root MuiGrid-container MuiGrid-wrap-xs-nowrap MuiGrid-justify-xs-space-between"
                    >
                      <p
                        class="MuiTypography-root makeStyles-defaultText-613 MuiTypography-body1"
                      >
                        Sildenafil
                         
                        50 mg
                        ,
                         
                        16 tablets
                      </p>
                      <p
                        class="MuiTypography-root makeStyles-defaultText-613 MuiTypography-body1"
                        style="padding-left: 10px;"
                      >
                        £123.00
                      </p>
                    </div>
                    <div
                      class="makeStyles-spacing1-638"
                      data-testid="spacing/root"
                    />
                  </div>
                </div>
              </div>
              <div
                class="MuiBox-root MuiBox-root-663"
              >
                <div
                  class="MuiBox-root MuiBox-root-664"
                >
                  <div
                    class="MuiBox-root MuiBox-root-665"
                  >
                    <h6
                      class="MuiTypography-root makeStyles-defaultText-613 MuiTypography-subtitle1"
                    >
                      Subscription discount
                    </h6>
                    <div
                      class="makeStyles-spacing1-638"
                      data-testid="spacing/root"
                    />
                    <div
                      class="makeStyles-discountRow-621"
                    >
                      <div
                        class="MuiBox-root MuiBox-root-666"
                      >
                        <svg
                          aria-hidden="true"
                          class="MuiSvgIcon-root makeStyles-promoIcon-611 MuiSvgIcon-fontSizeSmall"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M21.41 11.58l-9-9C12.05 2.22 11.55 2 11 2H4c-1.1 0-2 .9-2 2v7c0 .55.22 1.05.59 1.42l9 9c.36.36.86.58 1.41.58.55 0 1.05-.22 1.41-.59l7-7c.37-.36.59-.86.59-1.41 0-.55-.23-1.06-.59-1.42zM5.5 7C4.67 7 4 6.33 4 5.5S4.67 4 5.5 4 7 4.67 7 5.5 6.33 7 5.5 7z"
                          />
                        </svg>
                        <span
                          class="MuiTypography-root makeStyles-detailsText-614 MuiTypography-subtitle2"
                        >
                          Monthly saving
                        </span>
                      </div>
                      <p
                        class="MuiTypography-root makeStyles-discountValue-622 MuiTypography-body1"
                      >
                        <span
                          class="makeStyles-detailsText-614"
                        >
                          -
                          £5.20
                        </span>
                        <span
                          class="makeStyles-iconPlaceholder-631"
                        />
                      </p>
                    </div>
                    <div
                      class="makeStyles-spacing4-644"
                      data-testid="spacing/root"
                    />
                  </div>
                  <div
                    class="MuiFormControl-root MuiTextField-root makeStyles-root-667 MuiFormControl-fullWidth"
                  >
                    <label
                      class="MuiFormLabel-root MuiInputLabel-root makeStyles-label-673 MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined makeStyles-labelOutlined-674"
                      data-shrink="false"
                    >
                      Enter new code
                    </label>
                    <div
                      class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
                    >
                      <input
                        aria-invalid="false"
                        class="MuiInputBase-input MuiOutlinedInput-input makeStyles-inputText-670"
                        data-testid="basket/discount-code-input"
                        type="text"
                        value=""
                      />
                      <fieldset
                        aria-hidden="true"
                        class="PrivateNotchedOutline-root-676 MuiOutlinedInput-notchedOutline"
                      >
                        <legend
                          class="PrivateNotchedOutline-legendLabelled-678"
                        >
                          <span>
                            Enter new code
                          </span>
                        </legend>
                      </fieldset>
                    </div>
                  </div>
                  <div
                    class="MuiBox-root MuiBox-root-680"
                  >
                    <button
                      class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textSizeSmall MuiButton-sizeSmall MuiButton-fullWidth"
                      tabindex="0"
                      type="button"
                    >
                      <span
                        class="MuiButton-label"
                      >
                        <div
                          class="makeStyles-children-683"
                        >
                          <p
                            class="MuiTypography-root MuiTypography-body1"
                            style="color: rgb(0, 97, 242);"
                          >
                            Apply discount
                          </p>
                        </div>
                      </span>
                      <span
                        class="MuiTouchRipple-root"
                      />
                    </button>
                  </div>
                </div>
              </div>
              <div
                class="MuiBox-root MuiBox-root-686"
              >
                <hr
                  class="MuiDivider-root makeStyles-dividerDark-616"
                />
              </div>
              <div
                class="MuiBox-root MuiBox-root-687 makeStyles-totalWrapper-619"
              >
                <div
                  class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1 MuiGrid-align-items-xs-center"
                >
                  <div
                    class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6"
                  >
                    <h4
                      class="MuiTypography-root makeStyles-darkColor-617 MuiTypography-h4"
                      style="font-weight: 700;"
                    >
                      Total to pay
                    </h4>
                  </div>
                  <div
                    class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6"
                  >
                    <h2
                      class="MuiTypography-root makeStyles-darkColor-617 makeStyles-totalPrice-620 MuiTypography-h2 MuiTypography-alignRight"
                    >
                      £49.99
                    </h2>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
